gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/CodeableConceptMapper.java

```diff
@@ -0,0 +1,59 @@
+package uk.nhs.adaptors.pss.translator.mapper;
+
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.v3.CD;
+
+public class CodeableConceptMapper {
+    private static final String SNOMED_SYSTEM_CODE = "2.16.840.1.113883.2.1.3.2.4.15";
+    private static final String SNOMED_SYSTEM = "http://snomed.info/sct";
+
+    public CodeableConcept mapToCodeableConcept(CD codedData) {
+        if (hasSnomedCodeInMainCode(codedData)) {
+            return createCodeableConcept(codedData.getCode(), SNOMED_SYSTEM, codedData.getDisplayName(), codedData.getOriginalText());
+        } else if (hasSnomedCodeInTranslationElement(codedData)) {
+            var translation = getSnomedTranslationElement(codedData);
+            var display = getDisplayForSnomedTranslationElement(codedData, translation);
+            return createCodeableConcept(translation.getCode(), SNOMED_SYSTEM, display, codedData.getOriginalText());
+        }
+
+        var text = getText(codedData);
+        return new CodeableConcept().setText(text);
+    }
+
+    private boolean hasSnomedCodeInMainCode(CD codedData) {
+        return codedData.getCodeSystem().equals(SNOMED_SYSTEM_CODE);
+    }
+
+    private boolean hasSnomedCodeInTranslationElement(CD codedData) {
+        return codedData.getTranslation()
+            .stream()
+            .anyMatch(t -> t.getCodeSystem().equals(SNOMED_SYSTEM_CODE));
```

Comment:
More of a question here, but are we allowed to use single character variables like 't' in this case? For example would changing that to 'translation' be overkill?

Reply:
I've changed it to "translation", it will be more readable this way :)